easybuild/tools/filetools.py

```diff
@@ -1279,14 +1279,23 @@ def apply_patch(patch_file, dest, fn=None, copy=False, level=None, use_git_am=Fa
     return True
 
 
-def apply_regex_substitutions(path, regex_subs, backup='.orig.eb'):
+def apply_regex_substitutions(path, regex_subs, backup='.orig.eb', on_missing_match=None):
     """
     Apply specified list of regex substitutions.
 
     :param path: path to file to patch
     :param regex_subs: list of substitutions to apply, specified as (<regexp pattern>, <replacement string>)
     :param backup: create backup of original file with specified suffix (no backup if value evaluates to False)
+    :param on_missing_match: Define what to do when the file when no match was found in the file.
+                             Can be 'error' to raise an error, 'warn' to print a warning or 'ignore' to do nothing
+                             Defaults to value of --strict
     """
+    if on_missing_match is None:
+        on_missing_match = build_option('strict')
+    ALLOWED_VALUES = (run.ERROR, run.WARN, run.IGNORE)
+    if on_missing_match not in ALLOWED_VALUES:
+        raise EasyBuildError('Invalid value passed to on_missing_match: %s', on_missing_match)
```

Comment:
Also mention allowed values:

```suggestion
        raise EasyBuildError('Invalid value passed to on_missing_match: %s (allowed: %s)', on_missing_match, ', '.join(allowed_values))
```